mithril-stm/src/stm.rs

```diff
@@ -1323,4 +1498,85 @@ mod tests {
             })
         }
     }
+
+    //------------------------------------------------//
+    //----------------- Core Verifier -----------------//
+    //------------------------------------------------//
+    fn setup_equal_core_parties(
+        params: StmParameters,
+        nparties: usize,
+    ) -> (Vec<StmSigner<D>>, Vec<(VerificationKey, Stake)>) {
+        let stake = vec![1; nparties];
+        setup_core_parties(params, stake)
+    }
+
+    fn setup_core_parties(
+        params: StmParameters,
+        stake: Vec<Stake>,
+    ) -> (Vec<StmSigner<D>>, Vec<(VerificationKey, Stake)>) {
+        let mut trng = TestRng::deterministic_rng(ChaCha);
+        let mut rng = ChaCha20Rng::from_seed(trng.gen());
+
+        #[allow(clippy::needless_collect)]
```

Comment:
So we need this? Clippy does not complain in my machine if we don't use it, and if the collect is really not necessary, we could remove it.